### PR TITLE
 Updated base service interface dependency and documentation fixes

### PR DESCRIPTION
**Pull request by CaseyODonnell92**

### CODE REVIEW

File: src/main/java/io/electrum/airtime/api/model/VoucherRequest.java [hunk 1]
```diff
@@ -84,6 +86,16 @@ public void setTenders(List<Tender> tenders) {
       this.tenders = tenders;
    }
 
+   @ApiModelProperty(required = false, value = "An array of payment methods used to pay for the transaction")
```
Comment (TimElectrum) [inline src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: people are going to get very confused between paymentMethods and tenders. Please be a bit more descriptive in your description of paymentMethods - make it clear that they should be provided if this request is meant to effect payment too while tenders are supplied after payment has been effected. There should never be both present at once

Comment (neilmackellar) [reply src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: How about this: "An array of payment methods to be used as payment for the transaction. This is used if payment is not tendered at the point of sale, but is effected through one or more calls to third party payment providers as part of the request. A PaymentMethod differs from a TenderType in that the former represents payment that still needs to be collected from a third party, whereas the latter represents payment that has already been collected at the point of sale."

Comment (neilmackellar) [reply src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: Not sure about Tim's assertion that there never should be both. What about if part of the payment is cash and the balance is a token or loyalty card?

Comment (CaseyODonnell92) [reply src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: I've made some changes based directly on this comment. Please let me know if they're acceptable.

File: src/main/java/io/electrum/airtime/api/model/VoucherRequest.java [hunk 2]
```diff
@@ -84,6 +88,18 @@ public void setTenders(List<Tender> tenders) {
       this.tenders = tenders;
    }
 
+   @ApiModelProperty(required = false, value = "An array of payment methods used to pay for the transaction. If a " +
+           "request is meant to effect payment, then payment methods should be listed. No payment methods should be " +
+           "listed if tenders are supplied, as tenders are supplied after payment is effected.")
```
Comment (neilmackellar) [inline src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: My suggestion would be: 

"An array of payment methods to be used as payment for the transaction. This is used if payment is not tendered at the point of sale, but is effected through one or more calls to third party payment providers as part of the request. A PaymentMethod differs from a TenderType in that the former represents payment that still needs to be collected from a third party, whereas the latter represents payment that has already been collected at the point of sale."

File: src/main/java/io/electrum/airtime/api/model/VoucherRequest.java [hunk 3]
```diff
@@ -74,7 +76,9 @@ public VoucherRequest tenders(List<Tender> tenders) {
       return this;
    }
 
-   @ApiModelProperty(required = false, value = "An array of tenders used to pay for the transaction")
+   @ApiModelProperty(required = false, value = "An array of tenders used to pay for the transaction. Tenders are " +
+           "supplied after payment has been effected. No tenders should be listed in this field " +
+           "if payment methods are provided.")
```
Comment (neilmackellar) [inline src/main/java/io/electrum/airtime/api/model/VoucherRequest.java]: I was making a comment about this, but it seems to have gone missing...

I'm not sure if I agree with the assertion that tenders and payment methods are mutually exclusive. What about if part of the payment is tendered as cash and the balance is retrieved from a token/loyalty account?

**Pull request merged**